### PR TITLE
Rename ContentProvider to ContentSource

### LINKED ISSUE #4346: Rename the `ContentProvider` model to `ContentSource`
## Problem

In #4238, a confusing situation was raised when we found that the items of the `ContentProvider` model actually correspond to **sources**, not providers. We want to correct this naming so the usage of this model is clear and safe.

## Description

Here I copy @obulat's [comment](https://github.com/WordPress/openverse/pull/4238#discussion_r1602708722) explaning the situation:

> Originally, we only had `source`. Then, the `provider` field was added to denote the provider of the metadata about the image [^1]. The PR that added the `provider` field, didn't touch the `ContentProvider` model, and this model wasn't updated later.
>
> `ContentProvider` currently only handles _sources_, not _providers_. You can check this if you go to https://openverse.org/sources - you'll see NASA in the table. This uses the `stats` endpoint that queries `sources`, not `providers`.
>
>All this is to say that I think we _can_ proceed with this PR, and exactly because we rely on `ContentProvider` to filter sources. I'll review the code again to check that this is true.
>
> ...
>
> Could we open a new issue (maybe even a Project proposal request?) to 
> - Rename the `ContentProvider` model to `ContentSource`
> - Create a new `ContentProvider` model for providers
> - surface provider stats in the API
> - use this view to add provider links to the frontend single result page
> There is some additional info in the cc-archive issues and PRs (https://github.com/cc-archive/cccatalog-api/pull/548/files#r444933371, https://github.com/cc-archive/cccatalog-api/pull/560#issue-654914001)
>
> By the way, some of the issues mention that a source can appear in several providers. I think we have a project for implementing this, this would reduce the number of duplicates we have, but we would need to find a way to handle a work that has several providers. 
>
>[^1]: https://github.com/cc-archive/cccatalog-api/issues/531

### PR DESCRIPTION
**Pull request by obulat**
## Fixes

Fixes #4346 by @krysal

## Description

This PR renames the `ContentProvider` model to `ContentSource` to correctly reflect what the model is used for.

Interestingly, we had an inconsistency in included/excluded source query. The included sources query (`?source=<source_name>`) was querying _sources_ in ES, but the excluded sources query (`?excluded_source=<source_name>) was querying _providers_.
This PR changes the excluded sources query to exclude _sources_, not _providers_, as I think this is the correct way, and the documented way. I don't think it will affect the results much, and would like to hear reviewer opinions.

## Testing Instructions

The CI should pass.
The API should behave as usual:
- http://localhost:50280/v1/images/stats - should return all image sources (which are synonymous with providers in the sample data)
- http://localhost:50280/v1/images?source=wikimedia should return only the Wikimedia images
- http://localhost:50280/v1/images?excluded_source=flickr should return images from sources other than Flickr

Also try setting "Hide content" to true for one of the sources in the admin, and then check that `/stats` endpoint does not contain this source anymore. Also, `?source=<hidden_source>` should not return any results.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like`Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.
- [ ] I ran the DAG documentation generator (`just catalog/generate-docs` for catalog
      PRs) or the media properties generator (`just catalog/generate-docs media-props`
      for the catalog or `just api/generate-docs` for the API) where applicable.

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: api/api/controllers/search_controller.py [hunk 1]
```diff
@@ -41,11 +41,8 @@
...
 SOURCE_CACHE_TIMEOUT = 60 * 60 * 4  # 4 hours
 FILTER_CACHE_TIMEOUT = 30
-FILTERED_PROVIDERS_CACHE_KEY = "filtered_providers"
-FILTERED_PROVIDERS_CACHE_VERSION = 2
-THUMBNAIL = "thumbnail"
-URL = "url"
-PROVIDER = "provider"
+FILTERED_SOURCES_CACHE_KEY = "filtered_sources"
+FILTERED_SOURCES_CACHE_VERSION = 1
 QUERY_SPECIAL_CHARACTER_ERROR = "Unescaped special characters are not allowed."
```
Comment (obulat) [inline api/api/controllers/search_controller.py:46]:
```suggestion
```

**Review by dhruvkb — APPROVED** [review]
LGTM!

The name of the migration (`0066_contentsource_delete_contentprovider.py`) is a bit misleading. I would recommend updating it to something like `0066_contentprovider_renamed_contentsource.py`.

File: api/api/migrations/0066_contentsource_delete_contentprovider.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ("api", "0065_throttledapplication_privileges"),
+    ]
+
+    operations = [
+        migrations.RenameModel(old_name="ContentProvider", new_name="ContentSource"),
```
[the remark this answers is not in the document]
Comment (obulat) [reply api/api/migrations/0066_contentsource_delete_contentprovider.py:13]: Yes, not changing the database table/field names was specifically the requirement in this issue. I had to work around the django makemigration mechanism to make it recognize that the table is not deleted or created. That's the reason for the migration file name confusion, thank you for catching that, @dhruvkb

**Review by krysal — APPROVED** [review]
LGTM! I tested the parameters, and they work well.

Curiously, I found that several Stocksnap creators in the sample data have their websites linked to their Flickr profiles.

File: api/api/models/models.py [hunk 3]
```diff
@@ -4,24 +4,26 @@
...
+        ``source_identifier`` as "wikimedia_images" or "wikimedia" and
+        ``source_name`` as "Wikimedia"
     """
 
-    provider_identifier = models.CharField(max_length=50, unique=True)
-    provider_name = models.CharField(max_length=250)
+    source_identifier = models.CharField(
+        max_length=50, unique=True, db_column="provider_identifier"
+    )
+    source_name = models.CharField(max_length=250, db_column="provider_name")
```
[the remark this answers is not in the document]
Comment (obulat) [reply api/api/models/models.py:26]: Django power ✨️

**Pull request merged**